### PR TITLE
First remove the draft, then send the message

### PR DESCRIPTION
**Pull request by Jikstra**
This feels a bit better then first sending the message, then the message pops up, then the draft disappears

### CODE REVIEW

**Comment by Simon-Laux** [comment]
as I said before the right way would be sending the draft message, then reloading

**Comment by Jikstra** [comment]
> as I said before the right way would be sending the draft message, then reloading, then the deletion is not necessary anymore. If it should still be "slow" then there is probably a bug elsewhere

So we need an ipc call ala sendDraftMessage(chatId) ?

**Comment by Simon-Laux** [comment]
> So we need an ipc call ala sendDraftMessage(chatId) ?

yes

**Comment by Jikstra** [comment]
Just talked with @flipsimon about this and we found a problem with the sendDraftMessage idea. As the setDraft is debounced and the core takes some time, it's possible that sendDraftMessage would send an old draft. I think it makes more sense to just use the freshest message value from the ComposerInputRef, first remove the draft, then send the message.

File: src/renderer/components/composer/Composer.tsx [hunk 1]
```diff
@@ -98,18 +98,18 @@ const Composer = forwardRef<
         log.debug(`Empty message: don't send it...`)
         return
       }
+      /* clear it here to make sure the draft is cleared */
+      await DeltaBackend.call('messageList.removeDraft', chatId)
+      // /* update the state to reflect the removed draft */
+      window.__reloadDraft && window.__reloadDraft()
```
Comment (Simon-Laux) [inline src/renderer/components/composer/Composer.tsx:104]: do this after sending it. or just remove the await before send

**Review by Simon-Laux — APPROVED** [review]

**Pull request merged**